pkg/kubeapiserver/options/authentication.go

```diff
@@ -51,7 +51,8 @@ type AnonymousAuthenticationOptions struct {
 }
 
 type BootstrapTokenAuthenticationOptions struct {
-	Allow bool
+	Enable                       bool
+	DeprecatedEnableExperimental bool
```

Comment:
don't spread flag-specific things into the options structs... keep the single bool, bind two flags to it, and use fs.MarkDeprecated on the old flag